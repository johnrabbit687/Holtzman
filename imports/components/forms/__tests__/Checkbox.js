import { mount } from 'enzyme';
import Checkbox from '../Checkbox.js';
import { getSingleSpecWrapper } from "../../../util/tests/data-spec.js";
import { mountToJson } from "enzyme-to-json";

const generateComponent = (additionalProps={}) => (
    <Checkbox {...additionalProps} />
);

it ("should render with minimal props", () => {
  let component = mount(generateComponent());
  expect(mountToJson(component)).toMatchSnapshot();
});

it ("should accept a default value", () => {
  let component = mount(generateComponent({
    defaultValue: true
  }));
  expect(mountToJson(component)).toMatchSnapshot();

  const inputProps = getSingleSpecWrapper(component, "input").props();

  expect(inputProps.defaultChecked).toEqual("checked");
});

it ("should disable input with disabled prop", () => {
  let component = mount(generateComponent({
    disabled: true,
  }));
  expect(mountToJson(component)).toMatchSnapshot();

  const inputProps = getSingleSpecWrapper(component, "input").props();

  expect(inputProps.disabled).toEqual(true);
});

it ("should add classes with classes prop", () => {
  let component = mount(generateComponent({
    classes: ["test1", "harambe"],
  }));
  let component2 = mount(generateComponent({
    classes: "test1 harambe",
  }));
  expect(mountToJson(component)).toMatchSnapshot();
  expect(mountToJson(component2)).toMatchSnapshot();

  const wrapper = getSingleSpecWrapper(component, "input-wrapper");
  const wrapper2 = getSingleSpecWrapper(component2, "input-wrapper");

  expect(wrapper.hasClass("harambe")).toEqual(true);
  expect(wrapper.hasClass("harambe")).toEqual(true);
});

/*
DONE  defaultValue: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.string,
      ]),
DEPR  status: PropTypes.string,
DONE  disabled: PropTypes.any, // eslint-disable-line
DEPR  validation: PropTypes.func,
      errorText: PropTypes.string,
DEPR  theme: PropTypes.string,
DEPR  type: PropTypes.string,
      error: PropTypes.any, // eslint-disable-line
      classes: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.array,
      ]),
      children: PropTypes.any, // eslint-disable-line
      id: PropTypes.string,
      label: PropTypes.string,
      name: PropTypes.string,
      inputClasses: PropTypes.array, // eslint-disable-line
      clicked: PropTypes.func,
      hideLabel: PropTypes.bool,
*/