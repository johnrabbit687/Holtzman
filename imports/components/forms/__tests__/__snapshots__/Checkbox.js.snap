exports[`test should accept a default value 1`] = `
<Checkbox
  defaultValue={true}>
  <div
    className="input checkbox"
    data-spec="input-wrapper">
    <h6
      className="soft-left push-half-left flush-bottom text-left float-left locked-top">
      <small />
    </h6>
    <input
      data-spec="input"
      defaultChecked="checked"
      style={
        Object {
          "width": 0
        }
      }
      type="checkbox" />
    <Label>
      <label
        style={Object {}} />
    </Label>
  </div>
</Checkbox>
`;

exports[`test should add classes with classes prop 1`] = `
<Checkbox
  classes={
    Array [
      "test1",
      "harambe"
    ]
  }>
  <div
    className="input checkbox test1 harambe"
    data-spec="input-wrapper">
    <h6
      className="soft-left push-half-left flush-bottom text-left float-left locked-top">
      <small />
    </h6>
    <input
      data-spec="input"
      defaultChecked=""
      style={
        Object {
          "width": 0
        }
      }
      type="checkbox" />
    <Label>
      <label
        style={Object {}} />
    </Label>
  </div>
</Checkbox>
`;

exports[`test should add classes with classes prop 2`] = `
<Checkbox
  classes="test1 harambe">
  <div
    className="input checkbox test1 harambe"
    data-spec="input-wrapper">
    <h6
      className="soft-left push-half-left flush-bottom text-left float-left locked-top">
      <small />
    </h6>
    <input
      data-spec="input"
      defaultChecked=""
      style={
        Object {
          "width": 0
        }
      }
      type="checkbox" />
    <Label>
      <label
        style={Object {}} />
    </Label>
  </div>
</Checkbox>
`;

exports[`test should disable input with disabled prop 1`] = `
<Checkbox
  disabled={true}>
  <div
    className="input checkbox"
    data-spec="input-wrapper">
    <h6
      className="soft-left push-half-left flush-bottom text-left float-left locked-top">
      <small />
    </h6>
    <input
      data-spec="input"
      defaultChecked=""
      disabled={true}
      style={
        Object {
          "width": 0
        }
      }
      type="checkbox" />
    <Label>
      <label
        style={Object {}} />
    </Label>
  </div>
</Checkbox>
`;

exports[`test should render with minimal props 1`] = `
<Checkbox>
  <div
    className="input checkbox"
    data-spec="input-wrapper">
    <h6
      className="soft-left push-half-left flush-bottom text-left float-left locked-top">
      <small />
    </h6>
    <input
      data-spec="input"
      defaultChecked=""
      style={
        Object {
          "width": 0
        }
      }
      type="checkbox" />
    <Label>
      <label
        style={Object {}} />
    </Label>
  </div>
</Checkbox>
`;
